packages/twenty-server/src/engine/api/graphql/workspace-query-builder/factories/foreign-data-wrapper-query.factory.ts

```diff
@@ -29,6 +42,14 @@ export class ForeignDataWrapperQueryFactory {
     return `CREATE USER MAPPING IF NOT EXISTS FOR CURRENT_USER SERVER "${foreignDataWrapperId}" OPTIONS (user '${userMappingOptions.username}', password '${userMappingOptions.password}')`;
   }
 
+  updateUserMapping(
+    foreignDataWrapperId: string,
+    userMappingOptions: UserMappingOptions,
+  ) {
+    // CURRENT_USER works for now since we are using only one user. But if we switch to a user per workspace, we need to change this.
+    return `ALTER USER MAPPING FOR CURRENT_USER SERVER "${foreignDataWrapperId}" OPTIONS (SET user '${userMappingOptions.username}', SET password '${userMappingOptions.password}')`;
```

Comment:
Here you assumes that all fields will always be present. But we should only update the fields that are defined